lib/resolvers.js

```diff
@@ -29,3 +30,17 @@ export const mapLinkList = fieldName => doc => {
     return linkItem;
   }).filter(l => l !== undefined);
 };
+
+export const mapScheduleList = fieldName => doc => {
+  const schedule = doc[fieldName];
+
+  if (!schedule) { return []; }
+  return schedule.map((item) => {
```

Comment:
Between the original and the `reduce` example I posted we're likely to see the latter perform slower (a really trivial test using `window.performance` shows that it's about 7x slower on an array of 10000 elements). In this case I'd suggest performance trumps elegance as the code in its current form is still understandable, even if the undefined filtering seems superfluous compared to the equivalent reducer. 

I agree with @lpil - it would be interesting to see some proper perf on this one, because if we find one significantly out-performs the other but we're still looking at millions of operations then I'd go for the "elegant" solution and optimise if it becomes a bottleneck.